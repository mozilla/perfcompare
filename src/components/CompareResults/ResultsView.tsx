import { useEffect } from 'react';

import Grid from '@mui/material/Grid';
import { useLoaderData } from 'react-router-dom';
import { style } from 'typestyle';

import { useAppSelector } from '../../hooks/app';
import { SearchContainerStyles, background } from '../../styles';
import CompareWithBase from '../Search/CompareWithBase';
import { LinkToHome } from '../Shared/LinkToHome';
import PerfCompareHeader from '../Shared/PerfCompareHeader';
import type { LoaderReturnValue } from './loader';
import ResultsMain from './ResultsMain';

interface ResultsViewProps {
  title: string;
}
function ResultsView(props: ResultsViewProps) {
  const { baseRevInfo, newRevsInfo, frameworkId, results, baseRepo, newRepos } =
    useLoaderData() as LoaderReturnValue;

  const newRepo = newRepos[0];
  const { title } = props;
  const themeMode = useAppSelector((state) => state.theme.mode);
  const styles = {
    container: style({
      backgroundColor: background(themeMode),
    }),
  };

  const sectionStyles = SearchContainerStyles(themeMode, /* isHome */ false);

  useEffect(() => {
    document.title = title;
  }, [title]);

  return (
    <div
      className={styles.container}
      data-testid='beta-version-compare-results'
    >
      <PerfCompareHeader />
      <section className={sectionStyles.container}>
        <LinkToHome />

        <CompareWithBase
          hasNonEditableState={true}
          baseRev={baseRevInfo ?? null}
          newRevs={newRevsInfo ?? []}
          frameworkIdVal={frameworkId}
          isBaseSearch={null}
          expandBaseComponent={() => null}
          baseRepo={baseRepo}
          newRepo={newRepo}
        />
      </section>
      <Grid container alignItems='center' justifyContent='center'>
        <Grid item xs={12}>
          <ResultsMain results={results} />
        </Grid>
      </Grid>
    </div>
  );
}

export default ResultsView;
