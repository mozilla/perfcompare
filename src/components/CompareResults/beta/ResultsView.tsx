import { useEffect } from 'react';

import type { Theme } from '@mui/material';
import Grid from '@mui/material/Grid';
import { useLocation } from 'react-router-dom';
import { useSearchParams } from 'react-router-dom';
import { style } from 'typestyle';

import { compareView } from '../../../common/constants';
import { useAppDispatch, useAppSelector } from '../../../hooks/app';
import useFetchCompareResults from '../../../hooks/useFetchCompareResults';
import useHandleChangeSearch from '../../../hooks/useHandleChangeSearch';
import { comparisonResults as secondRevisionResults } from '../../../mockData/9d5066525489';
import { comparisonResults as thirdRevisionResults } from '../../../mockData/a998c42399a8';
import { comparisonResults as firstRevisionResults } from '../../../mockData/bb6a5e451dac';
import { setCompareData } from '../../../reducers/CompareResults';
import { SearchContainerStyles } from '../../../styles';
import { background } from '../../../styles';
import { fetchRecentRevisions } from '../../../thunks/searchThunk';
import { Repository, View, InputType } from '../../../types/state';
import CompareWithBase from '../../Search/CompareWithBase';
import PerfCompareHeader from '../../Shared/PerfCompareHeader';
import ResultsMain from './ResultsMain';

interface ResultsViewProps {
  protocolTheme: Theme;
  toggleColorMode: () => void;
  title: string;
}
function ResultsView(props: ResultsViewProps) {
  const dispatch = useAppDispatch();
  const repositoryBase = useAppSelector(
    (state) => state.search.base.repository,
  );
  const repositoryNew = useAppSelector((state) => state.search.new.repository);
  const { protocolTheme, toggleColorMode, title } = props;
  const themeMode = protocolTheme.palette.mode;
  const styles = {
    container: style({
      backgroundColor: background(themeMode),
    }),
  };

  const [searchParams] = useSearchParams();
  const fakeDataParam: string | null = searchParams.get('fakedata');

  const comparisonResults = {
    bb6a5e451dac: firstRevisionResults,
    '9d5066525489': secondRevisionResults,
    a998c42399a8: thirdRevisionResults,
  };

  // TODO: Populate store with real data or fake data pased on URL params
  useEffect(() => {
    if (fakeDataParam === 'true') {
      dispatch(setCompareData({ data: comparisonResults }));
    } else {
      dispatch(setCompareData({ data: {} }));
    }
  }, [fakeDataParam]);

  const sectionStyles = SearchContainerStyles(themeMode, compareView);

  const location = useLocation();
  const { dispatchFetchCompareResults } = useFetchCompareResults();
  const { searchByRevisionOrEmail } = useHandleChangeSearch();

  useEffect(() => {
    document.title = title;
  }, [title]);

  useEffect(() => {
    const urlSearchParams = new URLSearchParams(location.search);
    const repos = urlSearchParams.get('repos')?.split(',');
    const revs = urlSearchParams.get('revs')?.split(',');
    const framework = urlSearchParams.get('framework');

    if (revs && repos) {
      void dispatchFetchCompareResults(
        repos as Repository['name'][],
        revs,
        framework as string,
      );

      /*
      On component mount, use the repos and revs in hash to search for the base and new revisions. Store the results in state via the SelectedRevisionsSlice: see extra reducer, fetchRevisionsByID. Now can always display the selected revisions despite page refresh or copying and pasting url
      */
      revs.forEach((rev, index) => {
        void searchByRevisionOrEmail(
          repos[index] as Repository['name'],
          rev,
          'base',
        );
        void searchByRevisionOrEmail(
          repos[index] as Repository['name'],
          rev,
          'new',
        );
      });
    }
  }, []);

  /* editing the revisions requires fetching the recent revisions in results view
   */
  useEffect(() => {
    const repository = repositoryBase;
    void dispatch(
      fetchRecentRevisions({
        repository,
        searchType: 'base' as InputType,
      }),
    );
  }, [repositoryBase]);

  useEffect(() => {
    const repository = repositoryNew;
    void dispatch(
      fetchRecentRevisions({
        repository,
        searchType: 'new' as InputType,
      }),
    );
  }, [repositoryNew]);

  return (
    <div
      className={styles.container}
      data-testid='beta-version-compare-results'
    >
      <PerfCompareHeader
        themeMode={themeMode}
        toggleColorMode={toggleColorMode}
        view={compareView as View}
      />
      <section className={sectionStyles.container}>
        <CompareWithBase mode={themeMode} view={compareView as View} />
      </section>
      <Grid container alignItems='center' justifyContent='center'>
        <Grid item xs={12}>
          <ResultsMain themeMode={themeMode} />
        </Grid>
      </Grid>
    </div>
  );
}

export default ResultsView;
