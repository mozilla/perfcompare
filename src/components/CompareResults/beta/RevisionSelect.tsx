import SortIcon from '@mui/icons-material/Sort';
import Box from '@mui/material/Box';
import MenuItem from '@mui/material/MenuItem';
import Select from '@mui/material/Select';
import SvgIcon from '@mui/material/SvgIcon';
import { style } from 'typestyle'; 

const styles = {
  box: style({
    display: 'flex',
    gap: '1px',
  }),
  select: style({
    width: '300px',
  }),
};

const revisionsOptions = [
  'All revisions',
  '6064a73fc99f',
  'bc44c46861dd',
  '3bebf28f32a7',
];

function RevisionSelect() {
  return (
    <Box data-testid={'revision-select'}>
      <Select
        disabled // To be removed
        className={styles.select}
        defaultValue=''
        displayEmpty
        size='small'
        renderValue={(value) => {
          return (
            <Box className={styles.box}>
              <SvgIcon>
                <SortIcon />
              </SvgIcon>
              {value}
            </Box>
          );
        }}
      >
        {revisionsOptions.map((option) => (
          <MenuItem key={option} value={option}>
            {option}
          </MenuItem>
        ))}
      </Select>
    </Box>
  );
}


export default RevisionSelect;
