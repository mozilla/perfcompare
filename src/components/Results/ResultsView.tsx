import React from 'react';

import Container from '@mui/material/Container';
import Grid from '@mui/material/Grid';

import PerfCompareHeader from '../Shared/PerfCompareHeader';
import SelectedRevisionsTable from '../Shared/SelectedRevisionsTable';

function ResultsView() {
  return (
    <Container maxWidth="lg">
      <PerfCompareHeader />
      <Grid container>
        <Grid item xs={12}>
          <SelectedRevisionsTable />
        </Grid>
      </Grid>
    </Container>
  );
}

export default ResultsView;
