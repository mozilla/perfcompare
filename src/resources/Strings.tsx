export const Strings = {
  metaData: {
    pageTitle: {
      search: 'PerfCompare - Search',
      results: 'PerfCompare - Results',
    },
  },
  components: {
    topBanner: {
      text: 'This is an unstable pre-release version. Some features may not yet be supported.',
      linkText: 'File a bug on Bugzilla.',
      href: 'https://bugzilla.mozilla.org/enter_bug.cgi?product=Testing&component=PerfCompare&status_whiteboard=[pcf]',
    },
    header: {
      title: 'PerfCompare',
      tagline:
        'Analyze results of performance tests to detect regressions and identify opportunities for improvement.',
      bgLink:
        'url(https://user-images.githubusercontent.com/18633586/223226187-83621473-af10-4f7b-a7b9-564378b55c9e.png)',
      darkMode: 'Dark mode',
      lightMode: 'Light mode',
      learnMore: 'Learn more',
    },
    searchDefault: {
      title: 'Compare with a base or over time',
      sharedCollasped: {
        revisions: {
          selectLabel: 'Revisions',
          placeholder: 'Search base by ID number or author email',
        },
        framkework: {
          selectLabel: 'Framework',
          tooltip:
            'The framework or test harness containing the test you want to examine.',
        },
        button: 'Compare',
      },
      base: {
        title: 'Compare with a base',
        tagline: 'Analyze differences between specific revisions.',
        img: 'https://user-images.githubusercontent.com/88336547/233237125-1534220b-c343-421a-9133-ce8f151cb979.png',
        imgDark:
          'https://user-images.githubusercontent.com/88336547/233250674-004d071a-7c23-40f4-b348-0687a3fef6e3.png',

        save: 'Save',
        cancel: 'Cancel',
        editIcon:
          'https://user-images.githubusercontent.com/88336547/259900252-462ac221-f4ff-4b1c-bb6b-5df906e8007d.svg',
        collapsed: {
          warnings: {
            comparison:
              'Comparing “try” repository to any repository aside from "try" is not recommended.',
          },
          base: {
            selectLabel: 'Base',
            inputPlaceholder: 'Search base by ID number or author email',
            tooltip:
              'The baseline revision (no changes) your revision will be compared against.',
          },
          revision: {
            selectLabel: 'Revisions',
            inputPlaceholder: 'Search revision by ID number or author email',
            tooltip:
              'Revisions (typically including your changes) to compare against the selected base revision.',
          },
        },
      },
      overTime: {
        title: 'Compare over time',
        tagline:
          'Analyze differences between revisions within a specific time range.',
        img: 'https://user-images.githubusercontent.com/88336547/233250659-4012551b-e07a-44ce-accb-242e29d31914.png',
        imgDark:
          'https://user-images.githubusercontent.com/88336547/233250642-7fd7c217-e72b-4375-9078-7ed2f99cb0f7.png',
        collapedSelectLabel: 'Time range',
      },
    },
    expandableRow: {
      singleRun: 'Only one run (consider more runs for greater confidence).',
      Low: "A value of 'low' suggests less confidence that there is a sustained, significant change between the two revisions.",
      Medium:
        "A value of 'med' indicates uncertainty that there is a significant change. If you haven't already, consider retriggering the job to be more sure.",
      High: "A value of 'high' indicates more confidence that there is a significant change, however you should check the historical record for the test by looking at the graph to be more sure (some noisy tests can provide inconsistent results).",
    },
    noResultsFound: {
      mainMessage: 'No results found',
      note: 'For the selected revision(s), no results when compared to the base revision.',
    },
    comparisonRevisionDropdown: {
      allRevisions: 'All revisions',
    },
  },
  errors: {
    warningText: 'Search must be a 12- or 40-character hash, or email address',
  },
};
