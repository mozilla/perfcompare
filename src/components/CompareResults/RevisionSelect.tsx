import SortIcon from '@mui/icons-material/Sort';
import Box from '@mui/material/Box';
import MenuItem from '@mui/material/MenuItem';
import Select from '@mui/material/Select';
import SvgIcon from '@mui/material/SvgIcon';
import { useDispatch } from 'react-redux';
import { useLoaderData } from 'react-router-dom';
import { style } from 'typestyle';

import { useAppSelector } from '../../hooks/app';
import { updateComparison } from '../../reducers/ComparisonSlice';
import { Strings } from '../../resources/Strings';
import { truncateHash } from '../../utils/helpers';
import type { LoaderReturnValue } from './loader';

const styles = {
  box: style({
    display: 'flex',
    gap: '1px',
  }),
  select: style({
    width: '100%',
  }),
};

const allRevisionsOption =
  Strings.components.comparisonRevisionDropdown.allRevisions;

function RevisionSelect() {
  const dispatch = useDispatch();
  const { newRevs } = useLoaderData() as LoaderReturnValue;
  const { activeComparison } = useAppSelector((state) => state.comparison);
  const handlerChangeComparison = (option: string) => {
    dispatch(updateComparison({ activeComparison: option }));
  };

  return (
    <Box data-testid={'revision-select'}>
      <Select
        className={styles.select}
        defaultValue=''
        displayEmpty
        size='small'
        renderValue={(value) => {
          return (
            <Box className={styles.box}>
              <SvgIcon>
                <SortIcon />
              </SvgIcon>
              {value === allRevisionsOption.key
                ? allRevisionsOption.text
                : truncateHash(value)}
            </Box>
          );
        }}
        onChange={(event) => handlerChangeComparison(event.target.value)}
        value={activeComparison}
      >
        <MenuItem value={allRevisionsOption.key}>
          {allRevisionsOption.text}
        </MenuItem>
        {newRevs.map((rev) => (
          <MenuItem key={rev} value={rev}>
            {truncateHash(rev)}
          </MenuItem>
        ))}
      </Select>
    </Box>
  );
}

export default RevisionSelect;
