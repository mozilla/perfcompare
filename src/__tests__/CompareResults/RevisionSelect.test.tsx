import type { ReactElement } from 'react';

import ResultsView from '../../components/CompareResults/ResultsView';
import RevisionSelect from '../../components/CompareResults/RevisionSelect';
import { Strings } from '../../resources/Strings';
import {
  fireEvent,
  renderWithRouter,
  screen,
  within,
} from '../utils/test-utils';

function renderWithRoute(component: ReactElement) {
  return renderWithRouter(component, {
    route: '/compare-results/?fakedata=true',
  });
}

describe('Revision select', () => {
  it('Should match snapshot', () => {
    renderWithRoute(<RevisionSelect />);

    expect(screen.getByTestId('revision-select')).toBeInTheDocument();
    expect(document.body).toMatchSnapshot();
  });

  it("Should change 'All revisions' option to bb6a5e451dac", async () => {
    renderWithRoute(<RevisionSelect />);

    let selectButton = await screen.findByRole('button');

    expect(selectButton).toHaveTextContent('All revisions');

    fireEvent.mouseDown(selectButton);

    const listbox = within(await screen.findByRole('listbox'));

    fireEvent.click(await listbox.findByText('bb6a5e451dac'));

    selectButton = await screen.findByRole('button');
    expect(selectButton).toHaveTextContent('bb6a5e451dac');
  });

  it('Should render results for bb6a5e451dac', async () => {
    const location = {
      ...window.location,
      search: '?fakedata=true',
    };
    Object.defineProperty(window, 'location', {
      writable: true,
      value: location,
    });
    const urlParams = new URLSearchParams(window.location.search);
    const fakedataParam = urlParams.get('fakedata');
    expect(fakedataParam).toBe('true');

    renderWithRoute(<ResultsView title={Strings.metaData.pageTitle.results} />);

    // check to display results for all revisions
    let firstRevisionHeaders = await screen.findAllByRole('link', {
      name: /bb6a5e451dac/,
    });
    expect(firstRevisionHeaders.length).toBe(8);

    const secondRevisionHeaders = await screen.findAllByRole('link', {
      name: /9d5066525489/,
    });
    expect(secondRevisionHeaders.length).toBe(7);

    const thirdRevisionHeaders = await screen.findAllByRole('link', {
      name: /a998c42399a8/,
    });
    expect(thirdRevisionHeaders.length).toBe(7);

    // change comparison to revision bb6a5e451dac
    const selectRevisionDropdown = within(
      await screen.findByTestId('revision-select'),
    );
    let selectButton = await selectRevisionDropdown.findByRole('button');
    expect(selectButton).toHaveTextContent('All revisions');

    fireEvent.mouseDown(selectButton);

    const listbox = within(await screen.findByRole('listbox'));

    fireEvent.click(await listbox.findByText('bb6a5e451dac'));

    selectButton = await selectRevisionDropdown.findByRole('button');
    expect(selectButton).toHaveTextContent('bb6a5e451dac');

    fireEvent.mouseDown(selectButton);

    // check to display results only for revision bb6a5e451dac
    // findByRole doesn't work anymore because the root element
    // has aria-hidden but we don't know why yet
    firstRevisionHeaders = await screen.findAllByText(/bb6a5e451dac/, {
      selector: 'a',
    });
    expect(firstRevisionHeaders.length).toBe(8);

    expect(
      screen.queryAllByText(/9d5066525489/, {
        selector: 'a',
      }),
    ).toStrictEqual([]);

    expect(
      screen.queryAllByText(/a998c42399a8/, {
        selector: 'a',
      }),
    ).toStrictEqual([]);
  });

  it('Should render results for 9d5066525489', async () => {
    const location = {
      ...window.location,
      search: '?fakedata=true',
    };
    Object.defineProperty(window, 'location', {
      writable: true,
      value: location,
    });
    const urlParams = new URLSearchParams(window.location.search);
    const fakedataParam = urlParams.get('fakedata');
    expect(fakedataParam).toBe('true');

    renderWithRoute(<ResultsView title={Strings.metaData.pageTitle.results} />);

    // check comparison to be for revision bb6a5e451dac
    const firstRevisionHeaders = await screen.findAllByRole('link', {
      name: /bb6a5e451dac/,
    });
    expect(firstRevisionHeaders.length).toBe(8);

    // change comparison to revision 9d5066525489
    const selectRevisionDropdown = within(
      await screen.findByTestId('revision-select'),
    );
    let selectButton = await selectRevisionDropdown.findByRole('button');

    fireEvent.mouseDown(selectButton);

    const listbox = within(await screen.findByRole('listbox'));

    fireEvent.click(await listbox.findByText('9d5066525489'));

    selectButton = await selectRevisionDropdown.findByRole('button');
    expect(selectButton).toHaveTextContent('9d5066525489');

    fireEvent.mouseDown(selectButton);

    // check to display results only for revision 9d5066525489
    // findByRole doesn't work anymore because the root element
    // has aria-hidden but we don't know why yet
    const secondRevisionHeaders = await screen.findAllByText(/9d5066525489/, {
      selector: 'a',
    });
    expect(secondRevisionHeaders.length).toBe(7);

    expect(
      screen.queryAllByText(/bb6a5e451dac/, {
        selector: 'a',
      }),
    ).toStrictEqual([]);

    expect(
      screen.queryAllByText(/a998c42399a8/, {
        selector: 'a',
      }),
    ).toStrictEqual([]);
  });

  it('Should render results for a998c42399a8', async () => {
    const location = {
      ...window.location,
      search: '?fakedata=true',
    };
    Object.defineProperty(window, 'location', {
      writable: true,
      value: location,
    });
    const urlParams = new URLSearchParams(window.location.search);
    const fakedataParam = urlParams.get('fakedata');
    expect(fakedataParam).toBe('true');

    renderWithRoute(<ResultsView title={Strings.metaData.pageTitle.results} />);

    // check comparison to be for revision bb6a5e451dac
    const firstRevisionHeaders = await screen.findAllByRole('link', {
      name: /bb6a5e451dac/,
    });
    expect(firstRevisionHeaders.length).toBe(8);

    // change comparison to revision a998c42399a8
    const selectRevisionDropdown = within(
      await screen.findByTestId('revision-select'),
    );
    let selectButton = await selectRevisionDropdown.findByRole('button');

    fireEvent.mouseDown(selectButton);

    const listbox = within(await screen.findByRole('listbox'));

    fireEvent.click(await listbox.findByText('a998c42399a8'));

    selectButton = await selectRevisionDropdown.findByRole('button');
    expect(selectButton).toHaveTextContent('a998c42399a8');

    fireEvent.mouseDown(selectButton);

    // check to display results only for revision a998c42399a8
    // findByRole doesn't work anymore because the root element
    // has aria-hidden but we don't know why yet
    const thirdRevisionHeaders = await screen.findAllByText(/a998c42399a8/, {
      selector: 'a',
    });
    expect(thirdRevisionHeaders.length).toBe(7);

    expect(
      screen.queryAllByText(/9d5066525489/, {
        selector: 'a',
      }),
    ).toStrictEqual([]);

    expect(
      screen.queryAllByText(/bb6a5e451dac/, {
        selector: 'a',
      }),
    ).toStrictEqual([]);
  });
});
