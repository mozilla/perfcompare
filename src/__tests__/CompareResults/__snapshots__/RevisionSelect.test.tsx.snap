// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Revision select Should match snapshot 1`] = `
<body>
  <div>
    <div
      class="MuiBox-root css-0"
      data-testid="revision-select"
    >
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-sizeSmall fiy3v8i css-1od5rak-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
      >
        <div
          aria-expanded="false"
          aria-haspopup="listbox"
          class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall css-114a80n-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
          role="button"
          tabindex="0"
        >
          <div
            class="fbc330 MuiBox-root css-0"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-fuz6rv-MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-fuz6rv-MuiSvgIcon-root"
                data-testid="SortIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M3 18h6v-2H3v2zM3 6v2h18V6H3zm0 7h12v-2H3v2z"
                />
              </svg>
            </svg>
            All revisions
          </div>
        </div>
        <input
          aria-hidden="true"
          class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
          tabindex="-1"
          value="all-revisions"
        />
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-z44uxh-MuiSvgIcon-root-MuiSelect-icon"
          data-testid="ArrowDropDownIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-ihdtdm"
          >
            <span
              class="notranslate"
            >
              â€‹
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
  </div>
</body>
`;
